packages/server/src/tests/header/headerSelection.ts

```diff
@@ -97,35 +93,56 @@ const supportersTest: HeaderTest = {
     ],
 };
 
-const getNonSupportersTest = (edition: Edition): HeaderTest => {
-    if (edition === 'UK') {
-        return nonSupportersTestUK;
-    } else if (edition === 'US') {
-        return nonSupportersTestUS();
-    }
-    return nonSupportersTestNonUK;
-};
+const hardcodedTests = [supportersTest, nonSupportersTestUK, nonSupportersTestNonUK];
 
-export const selectHeaderTest = (
+// Exported for Jest testing
+export const selectBestTest = (
     targeting: HeaderTargeting,
-): Promise<HeaderTestSelection | null> => {
-    const select = (): HeaderTest => {
-        if (targeting.showSupportMessaging) {
-            return getNonSupportersTest(targeting.edition);
-        } else {
-            return supportersTest;
+    allTests: HeaderTest[],
+): HeaderTestSelection | null => {
+    const { showSupportMessaging, countryCode } = targeting;
+
+    const selectedTest: HeaderTest | undefined = allTests.find(test => {
+        const { isOn, userCohort, locations } = test;
+
+        if (!isOn) {
+            return false;
         }
-    };
 
-    const test = select();
-    const variant = test.variants[targeting.mvtId % test.variants.length];
-    if (test && variant) {
-        return Promise.resolve({
-            test,
-            variant,
-            moduleName: header.name,
-            modulePathBuilder: variant.modulePathBuilder,
-        });
+        if (!audienceMatches(showSupportMessaging, userCohort)) {
+            return false;
+        }
+
+        const testCountryCode = countryCode != null ? countryCode.toUpperCase() : '';
```

Comment:
The `!= null` check against countryCode feels a bit accidental, as `countryCode` can be `undefined` here.
I think we get away with it here because we're not using `!==`, but it's not very idiomatic.
But - given that `inCountryGroups` takes a `countryCode?: string`, would it be better to move the `toUpperCase` into that function and remove this line? It's a good thing to check, might as well do it for the whole codebase.

Reply:
Agreed. Will move.